### PR TITLE
WIP PR - New reports data API endpoints

### PR DESCRIPTION
**Pull request by diegomillan**

### CODE REVIEW

File: lms/djangoapps/grades/api/urls.py [hunk 1]
```diff
@@ -18,4 +18,25 @@
         ),
         views.CourseGradingPolicy.as_view(), name='course_grading_policy'
     ),
+    url(
+        r'^v0/course_grade/{course_id}/reportbysection/$'.format(
```
Comment (jagonzalr) [inline lms/djangoapps/grades/api/urls.py]: split url name from `reportbysection` to `report_by_section`

File: lms/djangoapps/grades/api/urls.py [hunk 2]
```diff
@@ -18,4 +18,25 @@
...
+            course_id=settings.COURSE_ID_PATTERN,
+        ),
+        views.GradeReportBySection.as_view(), name='grade_course_report_by_section'
+    ),
+    url(r'^v0/course_grade/report/(?P<uuid>[-\w]+)/$',
+        views.GradeReporByTaskId.as_view(), name='grade_course_report_generated'
+    ),
+    url(
+        r'^v0/course_grade/{course_id}/reportbyassignmenttype/$'.format(
```
Comment (jagonzalr) [inline lms/djangoapps/grades/api/urls.py]: split url name from `reportbyassignmenttype` to `report_by_assignment_type`

File: lms/djangoapps/grades/api/urls.py [hunk 3]
```diff
@@ -18,4 +18,25 @@
...
+            course_id=settings.COURSE_ID_PATTERN,
+        ),
+        views.GradeReportByAssignmentType.as_view(), name='grade_course_report_by_assignment_type'
+    ),
+    url(
+        r'^v0/course_grade/{course_id}/reportenhancedproblemgrade/$'.format(
```
Comment (jagonzalr) [inline lms/djangoapps/grades/api/urls.py]: split url name from `reportenhancedproblemgrade` to `report_enhanced_problem_grade`

File: common/lib/xmodule/xmodule/graders.py [hunk 4]
```diff
@@ -437,7 +438,7 @@ def total_with_drops(breakdown, drop_count):
 
             if not self.hide_average:
                 breakdown.append({'percent': total_percent, 'label': total_label,
-                                  'detail': total_detail, 'category': self.category, 'prominent': True})
+                                  'detail': total_detail, 'category': self.category, 'prominent': True, 'only': True, 'subsection': subsection,})
```
Comment (jagonzalr) [inline common/lib/xmodule/xmodule/graders.py]: what does `'only': True` means?

File: lms/djangoapps/grades/api/views.py [hunk 5]
```diff
@@ -226,3 +239,161 @@ def get(self, request, course_id, **kwargs):
...
+@view_auth_classes()
+class GradeReportBySection(GenericAPIView):
+    """
+    **Use Case**
+
+        Get the course grade report by section.
+
+    **Example requests**:
+
+        GET api/grades/v0/course_grade/{course_id}/reportbysection/
```
Comment (jagonzalr) [inline lms/djangoapps/grades/api/views.py]: update api endpoint name

File: lms/djangoapps/grades/api/views.py [hunk 6]
```diff
@@ -226,3 +239,161 @@ def get(self, request, course_id, **kwargs):
...
+@view_auth_classes()
+class GradeReportByAssignmentType(GenericAPIView):
+    """
+    **Use Case**
+
+        Get the course grade report by assignment type.
+
+    **Example requests**:
+
+        GET api/grades/v0/course_grade/{course_id}/reportbyassignmenttype/
```
Comment (jagonzalr) [inline lms/djangoapps/grades/api/views.py]: update api endpoint name

File: lms/djangoapps/grades/api/views.py [hunk 7]
```diff
@@ -226,3 +239,161 @@ def get(self, request, course_id, **kwargs):
...
+@view_auth_classes()
+class GradeReportEnhancedProblemGrade(GenericAPIView):
+    """
+    **Use Case**
+
+        Get the course enhanced problem grade report.
+
+    **Example requests**:
+
+        GET api/grades/v0/course_grade/{course_id}/reportenhancedproblemgrade/
```
Comment (jagonzalr) [inline lms/djangoapps/grades/api/views.py]: update api endpoint name

File: lms/djangoapps/instructor_task/tasks.py [hunk 8]
```diff
@@ -298,3 +299,39 @@ def export_ora2_data(entry_id, xmodule_instance_args):
     action_name = ugettext_noop('generated')
     task_fn = partial(upload_ora2_data, xmodule_instance_args)
     return run_main_task(entry_id, task_fn, action_name)
+
+@task(base=BaseInstructorTask)  # pylint: disable=not-callable
```
Comment (jagonzalr) [inline lms/djangoapps/instructor_task/tasks.py]: can we create a generic function that does this tasks and accepts the action name like `section_report` or `assignment_type_report`?

File: lms/djangoapps/grades/api/views.py [hunk 9]
```diff
@@ -226,3 +239,161 @@ def get(self, request, course_id, **kwargs):
         if isinstance(course, Response):
             return course
         return Response(GradingPolicySerializer(course.raw_grader, many=True).data)
+
+
+@view_auth_classes()
+class GradeReportBySection(GenericAPIView):
```
Comment (jagonzalr) [inline lms/djangoapps/grades/api/views.py]: class `GradeReportBySection`, `GradeReporByTaskId` and `GradeReportEnhancedProblemGrade` get function is the same code. The only thing that changes is the `submit_report_type`. Can we have a generic function instead?

**Review by jagonzalr — CHANGES_REQUESTED** [review]

File: common/lib/xmodule/xmodule/graders.py (hunk 4 above)
Comment (Squirrel18) [reply common/lib/xmodule/xmodule/graders.py]: I already added a code comment about that.

**Pull request closed without merging**